tests/integration/test_task_pool.py

```diff
@@ -1408,32 +1681,40 @@ async def test_runahead_future_trigger(
     async with start(schd, level=logging.DEBUG):
         assert str(schd.pool.runahead_limit_point) == '20010103'
         schd.pool.release_runahead_tasks()
-        for itask in schd.pool.get_all_tasks():
+        for itask in schd.pool.get_tasks():
             schd.pool.spawn_on_output(itask, 'succeeded')
         # future trigger raises the limit by one cycle point
         assert str(schd.pool.runahead_limit_point) == '20010104'
 
 
-async def test_compute_runahead_against_task_state(
-    flow,
-    scheduler,
-    start,
-    monkeypatch,
-):
-    """For each task status check whether changing the oldest task
-    to that status will cause compute_runahead to make a change.
-    """
-    states = [
+@pytest.mark.parametrize(
+    'status, expected',
+    [
         # (Status, Are we expecting an update?)
         (TASK_STATUS_WAITING, False),
         (TASK_STATUS_EXPIRED, True),
         (TASK_STATUS_PREPARING, False),
-        (TASK_STATUS_SUBMIT_FAILED, True),
+        (TASK_STATUS_SUBMIT_FAILED, False),
```

Comment:
Por que?

Reply:
This is a back-compat test. Cylc 7 ignored failed but not submit-failed tasks for runahead compute. (As commented in the test).

Reply:
Right, but the runahead logic [hasn't changed on this branch](https://github.com/cylc/cylc-flow/pull/5658/files#diff-becd22b70a92532763c24d9e0bedde95e36801c2eac01a469939f006c2faa148L295) so why has the test outcome changed?

Reply:
OK great spot 🥇 

On closer inspection this test is just completely screwed up. It is using task status strings as task messages, which are not always equivalent. I've confirmed that, at least in the submit-failed case, it wasn't doing what we thought (the "message" was actually being ignored as invalid).

I have a fix in the works, later tonight.

Reply:
@oliver-sanders - this has been addressed, I should have followed up to confirm my "later tonight" comment, sorry.

On master this test assumes that `failed` and `submit-failed` are treated the same way by Cylc 7, for runahead computation. In fact Cylc 7 does not do treat them the same way, but the test passes anyway because the test is broken - it erroneously sends TASK_STATUS values, instead of task output messages, to `process_message` with the result that the message isn't right in one case and it just gets ignored.

Now on this branch, the changed bool value correctly reflects the fact the Cylc 7 handles the two cases differently, and in the body of the task I use state reset instead of process_message, so that the original TASK_STATUS values are now appropriate.

Reply:
Ping @oliver-sanders - unresolved discussion